source/docs/example-projects/halloween-treasure-hunt.rst

```diff
@@ -0,0 +1,251 @@
+script container which, as we know, will listen to events happening within our server. We just have to give it a name, 
+like ``Halloween_Treasure_Hunt``, and set the ``type:`` key to ``world``. In addition to this, we'll also enable debug 
+by setting the ``debug:`` key to ``true``. This will make our script print in the console everything it does and help 
+us solve errors. It is not needed, so we can just disable it later on. We'll now add an ``events:`` subkey, which will 
+in the end hold the executable code we're going to write.
+
+Our file with the script container looks like this:
+
+.. code-block:: dscript
+    :name: figure_1_1
+    :linenos:
+    :emphasize-lines: 1-4
+
+    Halloween_Treasure_Hunt:
+    type: world
+    debug: true
+    events:
+
+.. rst-class:: figurecaption
+
+**Figure 1.1** Our starting world script container
+
+It's worth noting that Denizen scripts follow **YAML**'s indentation rules, with either *2 or 4-space tabs*. Many text 
+editors can be used for writing these scripts, but we personally prefer **Notepad++**. You just have to make sure the 
+tabs are replaced with actual spaces. There's a settings option in that automatically does that for us, so no problem.
+
+Now we need to find an event that fits our case. Looking through the *event list* documentation, ``on player clicks 
+block`` looks like our best bet. We just have to adjust it for our specific use case by adding *optional* arguments. 
+According to the event syntax, it accepts a click type, a material and an area, which will just be ``left``, 
+``pumpkin`` and ``in Hub`` respectively.
+
+.. note::
+    This tutorial assumes our *Hub world* is in fact called ``Hub``.
+
+We'll just make sure it works for now, so we add it under the events key and end the line with a ``:``. Inside this 
+event we can place our first command. A good option for testing purposes is just to show a word in *chat*. The command 
+for this is :guilabel:`narrate`. As the documentation explains, its syntax is very simple, only requiring a message 
+argument. We feel the hype, so we'll go with a ``- narrate "yay"``.
+
+Our script should be the following at this point:
+
+.. code-block:: dscript
+    :name: figure_1_2
+    :linenos:
+    :emphasize-lines: 5,6
+
+    Halloween_Treasure_Hunt:
+    type: world
+    debug: true
+    events:
+      on player left clicks pumpkin in Hub:
+      - narrate "yay"
+
+.. rst-class:: figurecaption
+
+**Figure 1.2** Our world script with a specific event
+
+It's time to *save* the script file, *reload* scripts ingame with ``/denizen reload scripts`` and *trigger the event* 
+by left clicking a pumpkin block in our Hub world. We should now be able to see a cute little ``yay`` in chat, along 
+with some debug information in the console, just as we expected. That's great, but we also have to *test* and make 
+sure the event is not being triggered when clicking other types of blocks, when right clicking, or when clicking in 
+another world.
```

Comment:
I feel that nothing in this paragraph is worth putting in italics.

For the last sentence here, you can probably just note that if the user *wants to be sure*, they can test to see that the event doesn't improperly trigger. The expectation is that the tutorial provides a working example that doesn't have an outstanding bug unless otherwise stated, so it's not really worth mentioning testing the script outside of a passing reference or telling the user that they should be debugging along the way.

Reply:
Taking the example doc as walking through what a user should be thinking and doing in the process of writing a script, I feel "don't need to test, of course the tutorial works" is basically 4th-wall breaking. We should absolutely encourage users as much as possible every time it's relevant to test their work. As we walk through what the user is thinking... at this stage it's "this all should work... but let's make sure".
Testing ones own code is never done as often as it should be (see for reference: half my Denizen commits >.>) and the more we encourage it the better, especially towards beginners (as would be reading through this rather basic level tutorial).

Reply:
Well yeah, you should mention testing the code thus far, but the general expectation is that the user should be testing along the way regardless. If needed, it can be included in the introductory section, where you tell the user to always test their code while they're making it to make sure it works.

On the other hand, I do see the worth in telling the user to test their code at this point. Though it should be more directed at the user, and not towards "us" (the authors *and* the user).